src/questions/questions.service.ts

```diff
@@ -1038,8 +1127,13 @@ export class QuestionsService {
     });
   }
 
-  async getQuestionCount(userId: number): Promise<number> {
-    return await this.questionRepository.countBy({ createdById: userId });
+  getQuestionCount(userId: number): Promise<number> {
+    return this.prismaService.question.count({
+      where: {
+        createdById: userId,
+        deletedAt: null,
```

Comment:
Is there really no need for a middleware here? As mentioned in the [Middleware sample: soft delete
](https://www.prisma.io/docs/orm/prisma-client/client-extensions/middleware/soft-delete-middleware#option-2-use-middleware-to-determine-the-behavior-of-readupdate-queries-for-deleted-records).

Reply:
accepted in commit https://github.com/SageSeekerSociety/cheese-backend/pull/226/commits/d87a714211f1dd9be5f758161c695eb681dea5d4